### PR TITLE
Logs RedisException message to error log on connection failure.

### PR DESCRIPTION
**Pull request by cnizzardini**
On a connection failure to the Redis server cake currently displays the following warning:

> Cache engine %s is not properly configured.

This is not particularly useful in determining the problem. This PR adds the RedisException message to the error log and informs the user to check there. Probably some better ways of going about this, but this is a step in the right direction IMO.

### CODE REVIEW

File: src/Cache/Engine/RedisEngine.php [hunk 1]
```diff
@@ -116,6 +117,8 @@ protected function _connect(): bool
                 );
             }
         } catch (RedisException $e) {
+            Log::error('RedisEngine RedisException: ' . $e->getMessage());
```
Comment (ADmad) [inline src/Cache/Engine/RedisEngine.php:120]: This needs to be wrapped in a `class_exists(Log::class)` check other wise we will have to add `cakephp/log` as a dependency of the `cakephp/cache` package.

File: src/Cache/CacheRegistry.php [hunk 2]
```diff
@@ -87,7 +87,10 @@ protected function _create($class, string $alias, array $config): CacheEngine
 
         if (!$instance->init($config)) {
             throw new RuntimeException(
-                sprintf('Cache engine %s is not properly configured.', get_class($instance))
+                sprintf(
+                    'Cache engine %s is not properly configured. Check log for additional information.',
```
Comment (ADmad) [inline src/Cache/CacheRegistry.php:91]:
```suggestion
                    'Cache engine %s is not properly configured. Check error log for additional information.',
```

**Comment by othercorey** [comment]
I think this is ok for master since it is just a log change.

File: src/Cache/Engine/RedisEngine.php (hunk 1 above)
Comment (cnizzardini) [reply src/Cache/Engine/RedisEngine.php:120]: Thanks. Should this use the FQN?

```php
            if (class_exists(\Cake\Log\Log::class)) {
                \Cake\Log\Log::error('RedisEngine RedisException: ' . $e->getMessage());
            }
```

Or is importing the class fine?

Comment (othercorey) [reply src/Cache/Engine/RedisEngine.php:120]: You can `use` the class normally as long as it's wrapped in class_exists since it might fail to load.

**Comment by cnizzardini** [comment]
Changing base to master creates conflict in travis.yaml. I don't feel comfortable changing that.

**Comment by othercorey** [comment]
> Changing base to master creates conflict in travis.yaml. I don't feel comfortable changing that.

You have to rebase the branch onto master then you can force-push and the merge will clean up.

File: src/Cache/Engine/RedisEngine.php [hunk 3]
```diff
@@ -116,6 +117,10 @@ protected function _connect(): bool
                 );
             }
         } catch (RedisException $e) {
+            if (class_exists(Log::class)) {
+                Log::error('RedisEngine RedisException: ' . $e->getMessage());
```
Comment (markstory) [inline src/Cache/Engine/RedisEngine.php:121]:
```suggestion
                Log::error('RedisEngine could not connect. Got error: ' . $e->getMessage());
```

Comment (cnizzardini) [reply src/Cache/Engine/RedisEngine.php:121]: Agreed, this is strictly related to establishing a connection.

**Pull request merged**